_extensions/custom_roles.py

```diff
@@ -19,12 +25,16 @@ def setup(app):
 
 def autolink(pattern):
     def role(name, rawtext, text, lineno, inliner, options={}, content=[]):
```

Comment:
options and content are only needed when custom in-document roles are used (`.. role:: new(base)`)

```suggestion
    def role(name, rawtext, text, lineno, inliner, _options=None, _content=None):
```